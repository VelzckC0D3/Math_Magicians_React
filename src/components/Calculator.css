.calculatorCont {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: fit-content;
  margin: 0 auto;
  border: 4px solid black;
  border-radius: 13px;
  background-color: white;
  box-shadow: -8px 10px 26px -8px rgba(0, 0, 0, 0.75);
}

.calculatorTitle {
  color: white;
  margin: 0 0 1rem;
  padding: 1rem 0;
  width: 22rem;
  background-color: black;
}

.calculator {
  width: 21rem;
  height: auto;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0.5rem;
}

.inputCont {
  width: 19rem;
  height: 3rem;
  text-align: right;
  background-color: rgb(216, 216, 216);
  color: rgb(0, 0, 0);
  font-size: 1.5rem;
  padding: 0 1rem;
  margin: 0 0 1rem;
  border-radius: 13px;
  border: 1px solid black;
}

.calculatorButton {
  padding: 1rem 2rem;
  width: 5rem;
  margin: 2px;
  transition: all 300ms ease-in-out;
  cursor: pointer;
  border-radius: 5px;
  font-size: 1.1rem;
  border: 1px solid black;
}

.calculatorButton:active {
  background-color: #cfcfcf;
  transition: all 50ms ease-in-out;
}

.mathButtonH {
  background-color: rgb(199, 199, 199);
}

.mathButtonH:active {
  background-color: #afafaf;
}

.mathButtonV {
  background-color: rgb(255, 132, 0);
}

.mathButtonV:active {
  background-color: rgb(179, 93, 0);
  transition: all 50ms ease-in-out;
}

.zero {
  width: 10.2rem;
}
