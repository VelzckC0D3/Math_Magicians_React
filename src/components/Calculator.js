import './Calculator.css';
import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => (
  <div className="calculatorCont">
    <h2 className="calculatorTitle">Calculator</h2>
    <ChildElement />
  </div>
);

function ChildElement() {
  const btnValues = [
    'AC', '+/-', '%', 'รท',
    '7', '8', '9', 'x',
    '4', '5', '6', '-',
    '1', '2', '3', '+',
    '0', '.', '=',
  ];

  const [result, setResult] = useState({
    total: 0,
    operation: null,
    next: null,
  });

  const handleButtonClick = (buttonName) => {
    setResult((prevState) => {
      const newResult = calculate(prevState, buttonName.target.textContent);
      return { ...prevState, ...newResult };
    });
  };

  return (
    <div className="calculator">
      <div className="calc-wrapper">
        <div className="inputCont">
          {result.total}
          {result.operation}
          {result.next}
        </div>
        <div className="rows">
          {btnValues.map((value) => (
            <button
              type="button"
              className="calculatorButton"
              onClick={handleButtonClick}
              key={value}
            >
              {value}
            </button>
          ))}
        </div>
      </div>
    </div>
  );
}

export default Calculator;
