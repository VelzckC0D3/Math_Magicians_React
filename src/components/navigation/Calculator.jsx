/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import calculate from '../logic/calculate';
import '../../style/Calculator.css';

const Calculator = () => {
  const [result, setResult] = useState({
    total: 0,
    operation: null,
    next: null,
  });

  const handleButtonClick = (buttonName) => {
    setResult((prevState) => {
      const newResult = calculate(prevState, buttonName.target.textContent);
      return { ...prevState, ...newResult };
    });
  };

  const btnValues = [
    'AC', '+/-', '%', 'รท',
    '7', '8', '9', 'x',
    '4', '5', '6', '-',
    '1', '2', '3', '+',
    '0', '.', '=',
  ];

  return (
    <div className="calculatorCont">
      <h2 className="calculatorTitle">Calculator</h2>
      <ChildElement
        result={result}
        handleButtonClick={handleButtonClick}
        btnValues={btnValues}
      />
    </div>
  );
};

function ChildElement({ result, handleButtonClick, btnValues }) {
  return (
    <div className="calculator">
      <div className="calc-wrapper">
        <div className="inputCont">
          {result.total}
          {result.operation}
          {result.next}
        </div>
        <div className="rows">
          {btnValues.map((value) => (
            <button
              type="button"
              className="calculatorButton"
              onClick={handleButtonClick}
              key={value}
            >
              {value}
            </button>
          ))}
        </div>
      </div>
    </div>
  );
}

export default Calculator;
