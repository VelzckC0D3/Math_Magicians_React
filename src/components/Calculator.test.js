import { render } from '@testing-library/react';
import Calculator from './calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container.querySelector('.inputCont')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
