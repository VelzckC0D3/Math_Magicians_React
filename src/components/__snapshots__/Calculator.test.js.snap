// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  class="calculatorCont"
>
  <h2
    class="calculatorTitle"
  >
    Calculator
  </h2>
  <div
    class="calculator"
  >
    <div
      class="calc-wrapper"
    >
      <div
        class="inputCont"
      >
        0
      </div>
      <div
        class="rows"
      >
        <button
          class="calculatorButton"
          type="button"
        >
          AC
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          +/-
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          %
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          รท
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          7
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          8
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          9
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          x
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          4
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          5
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          6
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          -
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          1
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          2
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          3
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          +
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          0
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          .
        </button>
        <button
          class="calculatorButton"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
